rest_docs/rest-catalog-open-api.yaml

```diff
@@ -0,0 +1,712 @@
+              }
+        404:
+          description: Not Found
+        406:
+          description: Not Acceptable - Unsupported Operation
+        422:
+          description: Unprocessable Entity - A property key was included in both removables and updates
+
+  /v1/namespaces/{namespace}/tables:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  identifiers:
+                    type: array
+                    items:
+                      $ref: '#/components/schemas/TableIdentifier'
+              examples:
+                TableIdentifierListExample:
+                  $ref: '#/components/examples/ListTablesExample'
+        404:
+          description: Not Found - Either the namespace or the table does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseErrorObject'
+              examples:
+                NamespaceNotFound:
+                  $ref: '#/components/examples/NoSuchNamespaceError'
+                TableNotFound:
+                  $ref: '#/components/examples/NoSuchTableError'
+
+  /v1/namespaces/{namespace}/tables/{table}:
+    parameters:
+      - $ref: '#/components/parameters/namespace'
+      - $ref: '#/components/parameters/table'
+    delete:
+      tags:
+        - Catalog API
+      summary: Drop a table from the catalog
+      operationId: dropTable
+      description: Remove a table from the catalog
+      parameters:
+        - name: purgeRequested
+          in: query
+          required: false
+          description: Whether the user requested to purge the underlying table's data and metadata
+          schema:
+            type: boolean
+            default: false
+      responses:
+        200:
+          description: OK
+          content:
+            'application/json':
+              schema:
+                $ref: '#/components/responses/DropTableResponse'
+              example: { "dropped": true, "purged": false }
+        202:
+          description: Accepted - for use if purgeRequested is implemented as an asynchronous action.
+        404:
+          $ref: '#/components/schemas/ResponseErrorObject'
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace.
+      responses:
+        200:
+          description: OK - Table Exists
+        401:
+          description: Unauthorized
+        404:
+          description: Not Found
```

Comment:
The rename route adds the schema and an example:

```
                TableToRenameDoesNotExist:
                  $ref: '#/components/examples/NoSuchTableError'
```

Should this be consistent and do the same here?

Reply:
So the way it is currently, 404 with just a plain description, signifies that there is no response object.

I'm happy to do that (with an example and schema) for all of them, it will just make the document a lot larger. But it is more correct.

Reply:
I found a nice middle ground and will be updating to that over the next 24-48 hours. =)

Reply:
OK. I've updated most everything to use non-inlined examples.

This section above uses `description` only to indicate that there is no response body (as it's a HEAD request).

We might choose to deviate from the standard and allow a response body on `HEAD` responses (like many systems do, including S3's API), at which time I'll update these from simple `Description` and status codes.